### ISSUE TITLE
DoHeatmap for ADT (v3)

### ISSUE THREAD

**Issue by joonlee3** [satijalab/seurat#927]
Hi,

   I was trying to create a heatmap with ADT markers by using Seurat v3. As shown in the figure attached below, I've got a single row named "NA" instead of the rows of ADT markers of interest. Do you provide any insights/suggestions for me to fix the problem? Thanks.

![image](https://user-images.githubusercontent.com/7217161/48300880-8b77dc80-e4b2-11e8-9028-570bd4556f79.png)

Best,
Joon

**Comment by satijalab** [comment]
HI Joon,

Thanks for trying out v3 and for your feedback.

1. Can you paste your call to DoHeatmap here? Please confirm that you either have your DefaultAssay set to ADT, or that you have specified the ADT assay in the DoHeatmap command.

2. Have you run ScaleData on your ADT values? We should issue a warning for this if ScaleData has not been run on features plotted in the heatmap, but do not as of yet,

**Comment by JBreunig** [comment]
v3 has been working very well for me as well.  Thanks for posting it! 

However, the DoHeatmap function, using the same call from v2 is giving similarly odd issues as Joon.  The colors can't be changed from purple, yellow, black and the ordering is off.  I will update later with my code and Heatmap from v2.3 and v3.

**Comment by GabeFalcao** [comment]
I had found a similar error @JBreunig. The issue is that there is no `genes.use` in the new function. Here is the help:  
`DoHeatmap(object, features = NULL, cells = NULL, group.by = "ident",
  group.bar = TRUE, disp.min = -2.5, disp.max = NULL,
  slot = "scale.data", assay = NULL, label = TRUE, size = 5.5,
  hjust = 0, angle = 45, combine = TRUE, ...)`  
You can use the `top50$gene` in the `features`, that worked for me.

**Comment by satijalab** [comment]
Thanks! Yes we have renamed some function arguments (in general, we tried to get rid of ".use", and often replaced genes with features to be more generalizable).

Because there is a ... in DoHeatmap you don't automatically see an error when using the wrong argument name. We're working on ways to make this more clear.

For what its worth, in v3 we tried our best to think carefully about how to name functions and arguments clearly. There are some changes now, but we expect more consistency in the long run.
